JS/Lesson19/esra/recursion.js

```diff
@@ -123,10 +185,73 @@ const nestedObjEx7 = {
 	},
 	h: 5,
 };
+
+function findMaxDepth(obj) {
+if (typeof obj !== 'object') {
+	return 0;
+}
+
+let maxDepth = 0;
+
+for (const key in obj) {
+	if (obj.hasOwnProperty(key)) {
+		const depth = findMaxDepth(obj[key]) + 1;
+		maxDepth = Math.max(maxDepth, depth);
+	}
+}
+
+return maxDepth;
+}
+
+const maxDepth = findMaxDepth(nestedObjEx7);
+console.log('Exercise7: ', 'Max Depth:', maxDepth);
+
 // 8. Write a recursive function to reverse the order of words in a sentence.
 const sentenceEx8 = 'Hello, how are you?';
+
+function reverseToWords(sentence) {
+	if (sentence.endsWith(' ')) {
+		sentence = sentence.trim() + ' ';
+	}
+
+	const firstSpaceIndex = sentence.indexOf(' ');
+
+	if (firstSpaceIndex === -1) {
+		return sentence;
+	} 
+
+	return reverseToWords(sentence.substring(firstSpaceIndex + 1)) + ' ' + sentence.substring(0, firstSpaceIndex);
+
+}
+
+const reversedSentence = reverseToWords(sentenceEx8);
+console.log('Exercise8: ', reversedSentence);
+
+
 // 9. Write a recursive function to find the length of the longest word in a sentence.
 const sentenceEx9 = 'The quick brown fox jumps over the lazy dog';
```

Comment:
Solution for this exercise is not correct, please try again:

1. Split the sentence into an array of words.
2. Define a recursive function that takes two parameters: the array of words and the current maximum length.
3. If the array is empty, return the maximum length.
4. Compare the length of the first word with the current maximum length and update it if necessary.
5. Recursively call the function with the rest of the array and the updated maximum length.